firestore/app/src/main/java/com/google/example/firestore/DocSnippets.java

```diff
@@ -1559,4 +1569,1387 @@ public void onComplete(@NonNull Task<AggregateQuerySnapshot> task) {
+        Task<Pipeline.Snapshot> results = db.pipeline()
+            .collection("books")
+            .aggregate(AggregateStage
+                .withAccumulators(
+                    AggregateFunction.average("rating").alias("avg_rating"))
+                .withGroups(Expression.field("genre")))
+            .execute();
+        // [END aggregate_groups]
+        System.out.println(results);
+    }
+
+    // https://cloud.google.com/firestore/docs/pipeline/overview#aggregate_distinct
+    void aggregateDistinct() {
+        // [START aggregate_distinct]
+        Task<Pipeline.Snapshot> results = db.pipeline()
+            .collection("books")
+            .distinct(
+                Expression.field("author").toUpper().alias("author"),
+                Expression.field("genre")
+            )
+            .execute();
+        // [END aggregate_distinct]
+        System.out.println(results);
+    }
+
+    // https://cloud.google.com/firestore/docs/pipeline/overview#sort
+    void sort() {
+        // [START sort]
+        Task<Pipeline.Snapshot> results = db.pipeline()
+            .collection("books")
+            .sort(
+                Expression.field("release_date").descending(),
+                Expression.field("author").ascending()
+            )
+            .execute();
+        // [END sort]
+        System.out.println(results);
+    }
+
+    // https://cloud.google.com/firestore/docs/pipeline/overview#sort
+    void sortComparison() {
+        // [START sort_comparison]
+        Query query = db.collection("cities")
+            .orderBy("state")
+            .orderBy("population", Query.Direction.DESCENDING);
+
+        Pipeline pipeline = db.pipeline()
+            .collection("books")
+            .sort(
+                Expression.field("release_date").descending(),
+                Expression.field("author").ascending()
+            );
+        // [END sort_comparison]
+        System.out.println(query);
+        System.out.println(pipeline);
+    }
+
+    // https://cloud.google.com/firestore/docs/pipeline/overview#functions
+    void functions() {
+        // [START functions_example]
+        Task<Pipeline.Snapshot> results;
+
+        // Type 1: Scalar (for use in non-aggregation stages)
+        // Example: Return the min store price for each book.
+        results = db.pipeline().collection("books")
+            .select(
+                Expression.field("current").logicalMinimum("updated").alias("price_min")
+            )
+            .execute();
+
+        // Type 2: Aggregation (for use in aggregate stages)
+        // Example: Return the min price of all books.
+        results = db.pipeline().collection("books")
+            .aggregate(AggregateFunction.minimum("price").alias("min_price"))
+            .execute();
+        // [END functions_example]
+        System.out.println(results);
+    }
+
+    // https://cloud.google.com/firestore/docs/pipeline/overview#creating_indexes
+    void creatingIndexes() {
```

Comment:
Why is this called creatingIndexes? I imagine the context text to explain this will be from difference source?